tests/integration/devfile/cmd_devfile_deploy_test.go

```diff
@@ -0,0 +1,45 @@
+package devfile
+
+import (
+	"path"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odo devfile deploy command tests", func() {
+
+	var commonVar helper.CommonVar
+
+	var _ = BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+		helper.Chdir(commonVar.Context)
+	})
+
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+
+	When("using a devfile.yaml containing a deploy command", func() {
+
+		BeforeEach(func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-deploy.yaml"), path.Join(commonVar.Context, "devfile.yaml"))
+			helper.Cmd("odo", "create").ShouldPass()
+		})
+		It("should run odo deploy", func() {
+			stdout := helper.Cmd("odo", "deploy").AddEnv("PODMAN_CMD=echo").ShouldPass().Out()
+			By("building and pushing image to registry", func() {
+				Expect(stdout).To(ContainSubstring("build -t quay.io/unknown-account/myimage -f " + commonVar.Context + "/Dockerfile " + commonVar.Context))
+				Expect(stdout).To(ContainSubstring("push quay.io/unknown-account/myimage"))
+			})
+			By("deploying a deployment with the built image", func() {
+				out := commonVar.CliRunner.Run("get", "deployment", "my-component", "-n", commonVar.Project, "-o", `jsonpath="{.spec.template.spec.containers[0].image}"`).Wait().Out.Contents()
+				Expect(out).To(ContainSubstring("quay.io/unknown-account/myimage"))
+			})
+		})
+	})
+})
```

Comment:
It looks like we aren't testing any specific success/failure scenarios here. Is it because `quay.io/unknown-account/myimage` can't be pushed and pulled during the test? Would it make sense to make this work by using a common quay account to which our CI can push and then pull from?

I'm just wondering how best we could test this.

Reply:
The main problem is that, for the moment, we cannot execute docker/podman on test environment because (most of the) tests are running on containers, and it is not easy to run `docker/podman build` on a container